pkg/provider/kubernetes/ingress/client.go

```diff
@@ -208,6 +209,9 @@ func (c *clientWrapper) WatchAll(namespaces []string, stopCh <-chan struct{}) (<
 		}
 	}
 
+	// Reset the stored server version to recheck on reconnection.
+	c.serverVersion = nil
```

Comment:
I'm not sure if this ever gets reset during the lifetime of Traefik once started.

Reply:
@dtomcej Moreover, I think we can put the logic of retrieving the server version in the `WatchAll` method, instead of the `GetServerVersion`, and then change `GetServerVersion` method to be only a "getter" method.
So we can remove the error from the signature.